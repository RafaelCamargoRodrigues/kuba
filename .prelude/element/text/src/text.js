import * as p from '@kuba/polished'
import styled from '@kuba/styled'

const text = new Proxy({}, {
  get (_target, key) {
    return styled[key]`
      color: ${p.color};
      font-family: ${p.family};
      font-size: ${p.size};
      font-weight: ${p.weight};
      line-height: var(--line-height-sm);
      text-align: ${p.align};
    `
  }
})

export default text
