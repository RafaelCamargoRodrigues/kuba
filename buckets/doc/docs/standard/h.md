---
sidebar_position: 2
---

# h
