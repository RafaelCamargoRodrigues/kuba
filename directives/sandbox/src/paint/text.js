import * as f from '@kuba/f'

export default {
  is (tag) {
    return f.equal(tag.type, 3)
  },

  execute (tag) {
    return tag.content
  }
}
