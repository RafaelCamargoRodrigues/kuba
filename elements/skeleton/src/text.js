import * as f from '@kuba/f'
import { size } from '@kuba/polished'
import styled from '@kuba/styled'

export default styled.span`
  background-color: var(--color-master-light);
  height: ${size};
  width: ${() => f.random(42, 89)}%;
`
