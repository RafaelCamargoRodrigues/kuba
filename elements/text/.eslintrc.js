module.exports = require('@kuba/eslintrc')