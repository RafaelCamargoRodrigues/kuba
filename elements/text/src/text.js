import { color, family, size, weight } from '@kuba/polished'
import styled from '@kuba/styled'

export default new Proxy({}, {
  get: (_, tag) =>
    styled[tag]`
      color: ${color};
      font-family: ${family};
      font-size: ${size};
      font-weight: ${weight};
      letter-spacing: 0.618px;
    `
})
