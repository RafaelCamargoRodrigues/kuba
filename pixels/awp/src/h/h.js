import * as f from '@rex/f'
import createNode from './createNode'
import createText from './createText'
import isNode from './isNode'
import isText from './isText'

export default f.cond(
  [isNode, createNode],
  [isText, createText]
)
