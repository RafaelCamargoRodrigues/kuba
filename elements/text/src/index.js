import skeleton from '@kuba/skeleton'
import notEmpty from './notEmpty'
import text from './text'

export default new Proxy({}, {
  get: (_, tagName) =>
    (props, children) =>
      notEmpty(children)
        ? text[tagName](props, children)
        : skeleton.text(props)
})
