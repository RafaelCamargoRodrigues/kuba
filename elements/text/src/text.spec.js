import h from '@kuba/h'
import text from '@kuba/text'

jest.mock('@kuba/h')

beforeEach(function () {
  h.mockClear()
})

test('O html tag eh determinado apos o operado de acesso', function () {
  text.P({}, []) // <text.P />
  
  expect(h).toHaveBeenCalled()
  expect(h).toHaveBeenCalledTimes(1)
  expect(h).toHaveBeenCalledWith('P', { className: [undefined, undefined] })
})

test('As cores e tamanhos sao determinados por atributos', function () {
  text.P({ master: true, darker: true, small: true }, []) // <text.P master darker small />

  expect(h).toHaveBeenCalled()
  expect(h).toHaveBeenCalledTimes(1)
  expect(h).toHaveBeenCalledWith('P', { master: true, darker: true, small: true, className: [undefined, undefined] })
})
