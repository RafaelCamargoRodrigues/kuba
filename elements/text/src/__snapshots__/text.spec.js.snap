// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar um paragrafo quando o namespace for text.P 1`] = `
<p
  style="color: var(--color-master); font-family: var(--font-family-base); font-size: var(--font-size-xs); font-weight: var(--font-weight-regular); letter-spacing: 0.618px;"
>
  kuba
</p>
`;
